pkg/oc/cli/admin/mustgather/mustgather.go

```diff
@@ -0,0 +1,247 @@
+		client config context, and then downloads the gathered information. 
+	`)
+
+	mustGatherExample = templates.Examples(`
+		# gather default information using the default image and command, writing into ./must-gather.local.<rand>
+		  oc adm must-gather
+
+		# gather default information with a specific local folder to copy to
+		  oc adm must-gather --dest-dir=/local/directory
+
+		# gather default information using a specific image, command, and pod-dir
+		  oc adm must-gather --image=my/image:tag --source-dir=/pod/directory -- myspecial-command.sh
+	`)
+)
+
+func NewMustGatherCommand(f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &Options{IOStreams: streams}
+	cmd := &cobra.Command{
+		Use:     "must-gather",
+		Short:   "Launch a new instance of a pod for gathering debug information",
+		Long:    mustGatherLong,
+		Example: mustGatherExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().StringVar(&o.NodeName, "node-name", "", "Set a specific node to use - by default a random master will be used")
+	cmd.Flags().StringVar(&o.Image, "image", "", "Set a specific image to use, by default the OpenShift's must-gather image will be used.")
+	cmd.Flags().StringVar(&o.DestDir, "dest-dir", "", "Set a specific directory on the local machine to write gathered data to.")
+
+	return cmd
+}
+
+func (o *Options) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	if i := cmd.ArgsLenAtDash(); i != -1 && i < len(args) {
+		o.Command = args[i:]
+	} else {
+		o.Command = args
+	}
+
+	var err error
+	o.ClientConfig, err = f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.KubeClient, err = kubernetes.NewForConfig(o.ClientConfig)
+	if err != nil {
+		return err
+	}
+
+	o.KubeConfig = f.ToRawKubeConfigLoader()
+
+	if len(o.Command) == 0 {
+		o.Command = []string{"/bin/sh"}
+	}
+
+	if len(o.DestDir) == 0 {
+		o.DestDir = fmt.Sprintf("must-gather.local.%06d", rand.Int63())
+	}
+
+	if len(o.Image) == 0 {
+		// TODO lookup cluster specific default
+		//		Image: "quay.io/openshift-release-dev/ocp-v4.0-art-dev:v4.0.5-1-ose-must-gather",
+		o.Image = "sanchezl/ocp-v4.0-art-dev:v4.0.5-1-ose-must-gather"
+	}
+
+	return nil
+}
+
+type Options struct {
+	ClientConfig *rest.Config
+	KubeClient   kubernetes.Interface
+
+	NodeName   string
+	DestDir    string
+	Image      string
+	Command    []string
+	KubeConfig clientcmd.ClientConfig
+
+	genericclioptions.IOStreams
+}
+
+// Run creates and runs a must-gather pod.d
+func (o *Options) Run() error {
+	if len(o.ClientConfig.BearerToken) != 0 {
+		return fmt.Errorf("cannot run must-gather with a token based kubeconfig")
+	}
+	if len(o.Image) == 0 {
+		return fmt.Errorf("missing an image")
+	}
+
+	// TODO This command will:
+	// [x] Create a temporary namespace
+	// [x] Create a secret an upload the local kubeconfig content into it
```

Comment:
let's not do this and grant an aggregator cluster-role to the service account in this namespace

Reply:
Simplest possible thing to start. Grant it cluster-admin to land

Reply:
Went with `cluster-admin` for now. Opened https://github.com/openshift/origin/pull/22430 as a reminder to fix in the future.